HomeWorkLinq/HomeWorkLinq/Program.cs

```diff
@@ -0,0 +1,145 @@
+                                         .Select(actor => actor)
+                                         .Count();
+
+            //4.Output two oldest actors names
+            var twoOldestActorsNames = data.Where(_ => _ is Film)
+                                           .SelectMany(_ => (_ as Film).Actors)
+                                           .OrderBy(actor => actor.Birthdate)
+                                           .Take(2)
+                                           .Select(actor => actor.Name)
+                                           .ToList();
+
+            //5.Output number of books per authors
+            var numberOfBooksPerAuthors = data.Where(_ => _ is Book)
+                                              .GroupBy(_ => (_ as Book).Author)
+                                              .Select(_ => new { NameAuthor = _.Key, CountOfBooks = _.Count() })
+                                              .ToList();
+
+            //6.Output number of books per authors and films per director
+            var numberOfBooksPerAuthorsAndFilmsPerDirector
+                        = data.Where(_ => _ is Book || _ is Film)
+                              .GroupBy(_ => _.GetType())
+                              .Select(_ => new { Author = _.Key, CountArts = _.Count() })
+                              .ToList();
+
+            //7.Output how many different letters used in all actors names
+            var howManyDifferentLettersInNames = data.Where(_ => _ is Film)
+                                     .SelectMany(_ => (_ as Film).Actors)
+                                     .SelectMany(actor => actor.Name)
+                                     .Distinct()
+                                     .Where(_ => _ != ' ')   // Is the space character a symbol?
+                                     .Count();
+
+            //8.Output names of all books ordered by names of their authors and number of pages
+
+            var namesBooksByAuthorsAndPages = data.Where(_ => _ is Book)
+                                                  .Select(_ => (_ as Book))
+                                                  .OrderBy(_ => _.Pages)
+                                                  .OrderBy(_ => _.Author)
+                                                  .Select(_ => $"{_.Name} - {_.Pages}")
+                                                  .ToList();
+
+            //9.Output actor name and all films with this actor
+            var actorNameAndAllFilmsWithHim = data.Where(_ => _ is Film)
+                                         .SelectMany(_ => (_ as Film).Actors)
+                                         .Where(actor => actor.Birthdate.Month == 8)
+                                         .Select(actor => actor)
+                                         .Count();
+
+            //10.Output sum of total number of pages in all books and all int values inside all sequences in data
+            var pages = data.Where(_ => _ is Book)
```

Comment:
I managed to get only the sum of the int array besides the sum of pages yet, not any sequence that holds int values... will spend some more time on it later.